sentry-android-core/src/main/java/io/sentry/android/core/PerformanceAndroidEventProcessor.java

```diff
@@ -99,13 +113,127 @@ public SentryEvent process(@NotNull SentryEvent event, @NotNull Hint hint) {
+      final @NotNull SentrySpan contentProviderRootSpan =
+          new SentrySpan(
+              contentProviderOnCreates.get(0).getStartTimestampSecs(),
+              contentProviderOnCreates
+                  .get(contentProviderOnCreates.size() - 1)
+                  .getProjectedStopTimestampSecs(),
+              traceId,
+              new SpanId(),
+              null,
+              UI_LOAD_OP,
+              "Content Providers",
+              SpanStatus.OK,
+              APP_METRICS_ORIGN,
+              new HashMap<>(),
+              null);
+      txn.getSpans().add(contentProviderRootSpan);
+      for (final @NotNull TimeSpan contentProvider : contentProviderOnCreates) {
+        txn.getSpans()
+            .add(
+                timeSpanToSentrySpan(
+                    contentProvider, contentProviderRootSpan.getSpanId(), traceId));
+      }
+    }
+
+    // Activities
+    final @NotNull List<ActivityLifecycleTimeSpan> activityLifecycleTimeSpans =
+        appStartMetrics.getActivityLifecycleTimeSpans();
+    if (!activityLifecycleTimeSpans.isEmpty()) {
+      final SentrySpan activityRootSpan =
+          new SentrySpan(
+              activityLifecycleTimeSpans.get(0).getOnCreate().getStartTimestampSecs(),
+              activityLifecycleTimeSpans
+                  .get(activityLifecycleTimeSpans.size() - 1)
+                  .getOnStart()
+                  .getProjectedStopTimestampSecs(),
+              traceId,
+              new SpanId(),
+              null,
+              UI_LOAD_OP,
+              "Activities",
+              SpanStatus.OK,
+              APP_METRICS_ORIGN,
+              new HashMap<>(),
+              null);
+      txn.getSpans().add(activityRootSpan);
+
+      for (ActivityLifecycleTimeSpan activityTimeSpan : activityLifecycleTimeSpans) {
+        if (activityTimeSpan.getOnCreate().hasStarted()
+            && activityTimeSpan.getOnCreate().hasStopped()) {
+          txn.getSpans()
+              .add(
+                  timeSpanToSentrySpan(
+                      activityTimeSpan.getOnCreate(), activityRootSpan.getSpanId(), traceId));
+        }
+        if (activityTimeSpan.getOnStart().hasStarted()
+            && activityTimeSpan.getOnStart().hasStopped()) {
+          txn.getSpans()
+              .add(
+                  timeSpanToSentrySpan(
+                      activityTimeSpan.getOnStart(), activityRootSpan.getSpanId(), traceId));
+        }
+      }
+    }
+  }
+
+  @NotNull
+  private static SentrySpan timeSpanToSentrySpan(
+      final @NotNull TimeSpan span,
+      final @Nullable SpanId parentSpanId,
+      final @NotNull SentryId traceId) {
+    return new SentrySpan(
+        span.getStartTimestampSecs(),
+        span.getProjectedStopTimestampSecs(),
+        traceId,
+        new SpanId(),
+        parentSpanId,
+        ActivityLifecycleIntegration.UI_LOAD_OP,
+        span.getDescription(),
+        SpanStatus.OK,
+        APP_METRICS_ORIGN,
+        new HashMap<>(),
```

Comment:
I guess it'd be cool to provide our users a way to attach custom tags or data for these spans, especially if they will be searchable in the future. But probably food for thought for the future (maybe could bring it up to the next starfish sync). Just thinking of a case where people would like to e.g. add custom tags to onCreate/onStart spans and then aggregate them by these tags (like a feature flag value).

They could do it in beforeSend of course, but maybe not always, e.g. some data is only available at time of execution, and if they could access current TimeSpan that'd be cool I guess.